### ISSUE TITLE
Add rule to disallow redundant 'use strict' directives

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#336]
Without type information this rule can only detect nested occurrences of `'use strict'`:

```ts
'use strict';
function foo() {
  'foo'; // should handle multiple prologue directives
  'use strict'; // already strict
}

function bar() {
  foo();
  'use strict'; // this is not a prologue directive and should be ignored -> will be handled by no-unused-expression
}
```

With type information this rule can also warn about all occurrences of `'use strict'` if `compilerOptions.alwaysStrict` is enabled.

**Comment by ajafff** [comment]
Note that certain ES2015 code is implicitly in strict mode: http://www.ecma-international.org/ecma-262/9.0/#sec-strict-mode-code

Also note that `'use\040strict'` is not a valid Use Strict Directive.

**Comment by ajafff** [comment]
It should also detect multiple Use Strict Directives in the same Directive Prologue:

```ts
'use strict';
'use strong';
'use strict'; // redundant
```
